org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/JdbcTemplate.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.consent.mgt.core.dao;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.helpers.MessageFormatter;
+import org.wso2.carbon.consent.mgt.core.constant.ConfigurationConstants;
+import org.wso2.carbon.consent.mgt.core.exception.DataAccessException;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A utility class to call JDBC with lambda expressions.
+ */
+public class JdbcTemplate {
+
+    private static final Logger logger = LoggerFactory.getLogger(JdbcTemplate.class);
+    private DataSource dataSource;
+
+    public JdbcTemplate(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    /**
+     * Executes a query on JDBC and return the result as a list of domain objects.
+     *
+     * @param query     the SQL query with the parameter placeholders.
```

Comment:
Shall we start a line with an upper case letter? Also, shall we fix other places as well?